### PR TITLE
add support libc++_shared on android (master)

### PR DESCRIPTION
**Pull request by rdhafidh**
Follow up https://github.com/godotengine/godot/pull/19908

### CODE REVIEW

**Comment by akien-mga** [comment]
It seems to trigger a build issue on Travis: https://travis-ci.org/godotengine/godot/jobs/408825831

Current SDK/NDK versions used on Travis are described here: https://github.com/godotengine/godot/blob/master/misc/travis/android-tools-linux.sh
We could upgrade to newer versions, but we need to keep compatibility with those relatively recent SDK/NDK versions anyway.

**Comment by rdhafidh** [comment]
Ah sorry I'll reproduce that locally, didn't catch this before

**Comment by rdhafidh** [comment]
Has been Figured out now on how to build with r15c, I'm still testing

**Comment by rdhafidh** [comment]
hopefully ci test will pass now

File: platform/android/detect.py [hunk 1]
```diff
@@ -203,6 +203,13 @@ def mySpawn(sh, escape, cmd, args, env):
     lib_sysroot = env["ANDROID_NDK_ROOT"] + "/platforms/" + env['ndk_platform'] + "/" + env['ARCH']
 
     ## Compile flags
+
```
Comment (akien-mga) [inline platform/android/detect.py]: Trailing spaces here.

File: platform/android/detect.py [hunk 2]
```diff
@@ -208,14 +208,15 @@ def mySpawn(sh, escape, cmd, args, env):
...
+        env.Append(CPPFLAGS=["--sysroot="+sysroot])
         env.Append(CPPFLAGS=["-isystem", sysroot + "/usr/include/" + abi_subpath])
+        env.Append(CPPFLAGS=["-isystem", env["ANDROID_NDK_ROOT"] + "/sources/android/support/include"])
         # For unified headers this define has to be set manually
         env.Append(CPPFLAGS=["-D__ANDROID_API__=" + str(get_platform(env['ndk_platform']))])
     else:
         print("Using NDK deprecated headers")
         env.Append(CPPFLAGS=["-isystem", lib_sysroot + "/usr/include"])
-
+
```
Comment (akien-mga) [inline platform/android/detect.py]: Trailing spaces.

File: platform/android/detect.py [hunk 3]
```diff
@@ -247,22 +248,30 @@ def mySpawn(sh, escape, cmd, args, env):
...
+        env.Append(CPPFLAGS=["-isystem", env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/llvm-libc++/include"])
+        env.Append(CPPFLAGS=["-isystem", env["ANDROID_NDK_ROOT"] + "/sources/cxx-stl/llvm-libc++abi/include"])
+        env.Append(CXXFLAGS=['-frtti',"-std=gnu++14"])
     else:
         env.Append(CXXFLAGS=['-fno-rtti', '-fno-exceptions', '-DNO_SAFE_CAST'])
 
     ## Link flags
-
-    env['LINKFLAGS'] = ['-shared', '--sysroot=' + lib_sysroot, '-Wl,--warn-shared-textrel']
+    if ndk_version != None and LooseVersion(ndk_version) >= LooseVersion("15.0.4075724"):
```
Comment (akien-mga) [inline platform/android/detect.py]: Could you add explanatory comments for why those two versions require specific treatment?

**Comment by akien-mga** [comment]
Looks fine overall, though check the few comments I put in review. If you can address those and squash the commits, it should be good to merge.

**Comment by akien-mga** [comment]
Ping.

**Comment by rdhafidh** [comment]
ah apologize, long time ago mostly forgot this one

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Already cherry-picked in #23033.

File: platform/android/detect.py (hunk 3 above)
Comment (jahd2602) [reply platform/android/detect.py]: @rdhafidh Indeed, a comment or two on that specific treatment would be useful.

**Pull request merged**